ArmaforcesMissionBot/Modules/Signups.cs

```diff
@@ -1,38 +1,73 @@
 using ArmaforcesMissionBot.Attributes;
 using ArmaforcesMissionBot.DataClasses;
-using ArmaforcesMissionBot.Handlers;
 using Discord;
 using Discord.Commands;
-using Discord.Rest;
 using Discord.WebSocket;
 using Microsoft.Extensions.DependencyInjection;
 using Newtonsoft.Json;
 using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
 using System.Net;
-using System.Text;
+using System.Net.Http;
+using System.Reflection;
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
+using ArmaforcesMissionBot.Exceptions;
 using ArmaforcesMissionBot.Extensions;
+using ArmaforcesMissionBot.Features;
+using ArmaforcesMissionBot.Features.Signups.Importer;
+using ArmaforcesMissionBot.Helpers;
 using ArmaforcesMissionBotSharedClasses;
 using static ArmaforcesMissionBot.DataClasses.OpenedDialogs;
 
 namespace ArmaforcesMissionBot.Modules
 {
     [Name("Zapisy")]
-    public class Signups : ModuleBase<SocketCommandContext>
+    public class Signups : ModuleBase<SocketCommandContext>, IModule
     {
         public IServiceProvider _map { get; set; }
         public DiscordSocketClient _client { get; set; }
         public Config _config { get; set; }
         public OpenedDialogs _dialogs { get; set; }
         public CommandService _commands { get; set; }
+        public SignupsData SignupsData { get; set; }
 
-        public Signups()
-        {
-            //_map = map;
+        [Command("importuj-zapisy")]
+        [Summary("Importuje zapisy z załączonego pliku *.txt. lub z wiadomości (preferując plik txt jeżeli obie rzeczy są). " +
+                 "Czyta plik/wiadomość linia po linii, dołączając linie bez prefixu 'AF!' do poprzedniej komendy " +
+                 "a następnie wywołuje komendy w kolejności. " +
+                 "Ignoruje linie zaczynające się od '#' oraz '//' umożliwiając komentarze.")]
+        [ContextDMOrChannel]
+        public async Task ImportSignups([Remainder]string missionContent = null) {
+            if (_client.GetGuild(_config.AFGuild)
+                .GetUser(Context.User.Id)
+                .Roles.All(x => x.Id != _config.MissionMakerRole))
```

Comment:
Shouldn't this be Roles.Any?

Reply:
It is ok, we need to check whether any user role is the mission maker role, and if not then return him an error. It's just inverted for less nesting.